scripts-components/in-view/src/index.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, ComponentInitializationError } from '@mangoweb/scripts-base'
+import { IsInView, isInView, isOutOfView, isSupported } from 'isinview'
+
+interface InViewData {
+	targets?: string
+	threshold?: number
+
+	isAboveViewClass?: string
+	isInViewClass?: string
+	isBelowViewClass?: string
+	isSeenClass?: string
+}
+
+export class InView extends Component<InViewData> {
+	static componentName = 'InView'
+
+	protected readonly targets: IsInView.Target
+	protected readonly options: Partial<IsInView.Options> = {}
+
+	protected readonly isAboveViewClass: string
+	protected readonly isInViewClass: string
+	protected readonly isBelowViewClass: string
+	protected readonly isSeenClass: string
+
+	public constructor(el: HTMLElement, data: InViewData) {
+		super(el, data)
+
+		if (!isSupported()) {
+			throw new ComponentInitializationError('InView component not supported. Add IntersectionObserver polyfill.')
+		}
+
+		this.targets = this.data.targets ? this.getChildren(this.data.targets) : this.el
+
+		if (this.data.threshold) {
+			this.options.threshold = this.data.threshold
+		}
+
+		this.isAboveViewClass = this.getProp('isAboveViewClass', 'is-aboveView')
+		this.isInViewClass = this.getProp('isInViewClass', 'is-inView')
+		this.isBelowViewClass = this.getProp('isBelowViewClass', 'is-belowView')
+		this.isSeenClass = this.getProp('isSeenClass', 'is-seen')
```

Comment:
@JirkaVebr Tohle jsem úplně nechytil během review scripts-base... Proč se to jmenuje `getProp` a ne `getData`? `prop` znamená `property`, což tady nesedí.

Reply:
Taky mi to uniklo. :(

Když už se tu otevírá diskuze, co si myslíte o tom, že by druhý argument mohl být libovolného typu a nepovinný? Trochu mě svádělo vůbec nesahat na `this.data` a všude používat jen `this.getProp`, aby jsem byl konzistentní a nepřistupoval k datům různými způsoby. (Možná libovolný typ je zbytečně moc liberální. Každopádně chyběl mi způsob, jak se pomocí `getProp` dostat k datům i s informací, že hodnota nebyla nastavena.)

Reply:
Současný stav není ideální, no. Správné řešení je podle mě `data` přejmenovat na `props`, mít `getProp` jako pouhý getter a pak `getPropOrElse` s povinným fallbackem. Jméno `getData` jsem záměrně nezvolil, protože je gramaticky špatně; správný singulár `getDatum` by ale zas byl možná matoucí. I proto soudím, že správné řešení je přejmenovat `data`. Je to ale samozřejmě BC break jako kráva, ale já jsem klidně pro.

Reply:
Líbí se mi Jirkův návrh. Jsem taky pro. IMO lepší to udělat dříve než později a tohle věřím, že udělat chceme. `data` => `props`.

Reply:
Souhlasím, že s `props` se více přiblížíme Reactu a Web components.

---

Off topic: Dle wikipedie je použití getData v pohodě https://en.m.wikipedia.org/wiki/Data_(word)

Reply:
Připravil jsem pull request: #13.

Ad to datum/data, podle mě je to spíš jen případ toho, že to tak dlouho všichni používali špatně, až to mnozí přestali považovat za špatné. I tak bych se tomu ale raději vyhnul.